ouroboros-consensus/src/Ouroboros/Storage/ChainDB/Impl/Background.hs

```diff
@@ -238,7 +238,7 @@ garbageCollect CDB{..} slotNo = do
     VolDB.garbageCollect cdbVolDB slotNo
     atomically $ do
       LgrDB.garbageCollectPrevApplied cdbLgrDB slotNo
-      modifyTVar' cdbInvalid $ Set.filter ((<= slotNo) . pointSlot)
+      modifyTVar' cdbInvalid $ Map.filter (> slotNo)
```

Comment:
The new tests with invalid blocks helped me discover this bug: instead of removing the points corresponding to *old* invalid blocks, we were removing the points corresponding to *new* invalid blocks.

Reply:
How frequently is this called, once per block or slot? The Map filter is linear. There's a trade-off for how frequently we scan vs how much excess memory we use.